src/telemetry/extension.js

```diff
@@ -229,6 +231,19 @@ export async function getBadgeType() {
   }
 }
 
+export async function hasBadgeTextChanged(nextText, tabId = null) {
```

Comment:
This API is difficult to read at the point of usage; `hasBadgeTextChanged('+')` doesn't really make it clear that the parameter is the badge text we're intending to change to.

Since it's being passed in anyway, maybe change this to `getToolbarBadgeText(tabId)` and do the comparison in place?

```js
if (getToolbarBadgeText(tabId) !== '+') {

}
```